lrtable/src/lib/statetable.rs

```diff
@@ -406,10 +412,14 @@ where
     /// Return the goto state for `stidx` and `ridx`, or `None` if there isn't any.
     pub fn goto(&self, stidx: StIdx, ridx: RIdx<StorageT>) -> Option<StIdx> {
         let off = (usize::from(stidx) * usize::from(self.rules_len)) + usize::from(ridx);
-        if self.gotos[off] == StIdx::max_value() {
+        // Goto entries are encoded by adding 1 to their value, while 0 is reserved for no entry
+        // (i.e. error)
+        if self.gotos.get(off).unwrap() == 0 {
```

Comment:
It would be good not to call `get` twice (the compiler *might* optimise this for us, but it's difficult to know). One solution might be along the lines of:

```rust
match self.gotos.get(off) {
    Some(0) => None,
    Some(i) => Some(StIdx((i - 1) as StIdxStorageT),
    None => unreachable!()
}
```

Reply:
Pushed changes!